### ISSUE TITLE
Dependencies from previous versions of module are included

### ISSUE THREAD

**Issue by thlapin** [unibet/puppet-forge-server#7]
This best example of this is to try and install the pupppetlabs-rabbitmq module. In an earlier version it had a dependency on puppetlabs-prosvc_repo. This module no longer exists.

```
$ puppet module install --module_repository=http://localhost:8080 puppetlabs-rabbitmq
Notice: Preparing to install into /Users/thlapin/.puppetlabs/etc/code/modules ...
Notice: Created target directory /Users/thlapin/.puppetlabs/etc/code/modules
Notice: Downloading from http://localhost:8080 ...
Error: Request to Puppet Forge failed.
  The server being queried was http://localhost:8080/v3/releases?module=puppetlabs-prosvc_repo
  The HTTP response we received was '404 Not Found'
```

**Comment by i11** [comment]
Nice catch! Thanks for reporting it! I'm going to fix it asap.

**Comment by i11** [comment]
Fixed with 1.4.2.

```
$ puppet module install puppetlabs-rabbitmq --debug --module_repository http://localhost:8080
Notice: Preparing to install into /Users/myuser/.puppet/modules ...
Notice: Created target directory /Users/myuser/.puppet/modules
Notice: Downloading from http://localhost:8080 ...
Debug: HTTP GET http://localhost:8080/v3/releases?module=puppetlabs-rabbitmq
Debug: HTTP GET http://localhost:8080/v3/releases?module=puppetlabs-prosvc_repo
Debug: HTTP GET http://localhost:8080/v3/releases?module=puppetlabs-stdlib
Debug: HTTP GET http://localhost:8080/v3/releases?module=puppetlabs-apt
Debug: HTTP GET http://localhost:8080/v3/releases?module=garethr-erlang
Debug: HTTP GET http://localhost:8080/v3/releases?module=stahnma-epel
Debug: HTTP GET http://localhost:8080/v3/releases?module=nanliu-staging
Info: Resolving dependencies ...
Info: Preparing to install ...
Debug: HTTP GET http://localhost:8080/v3/files/v3/files/puppetlabs-rabbitmq-5.1.0.tar.gz
Debug: Executing 'gzip -dc /Users/myuser/.puppet/var/puppet-module/cache/puppetlabs-rabbitmq20150417-52237-1izvyee | tar xof -'
Debug: Executing 'find . -type d -exec chmod 755 {} +'
Debug: Executing 'find . -type f -exec chmod a-wst {} +'
Debug: Executing 'chown -R 1018876844:1884119166 .'
Debug: HTTP GET http://localhost:8080/v3/files/v3/files/nanliu-staging-1.0.3.tar.gz
Debug: Executing 'gzip -dc /Users/myuser/.puppet/var/puppet-module/cache/nanliu-staging20150417-52237-1pk0ovs | tar xof -'
Debug: Executing 'find . -type d -exec chmod 755 {} +'
Debug: Executing 'find . -type f -exec chmod a-wst {} +'
Debug: Executing 'chown -R 1018876844:1884119166 .'
Debug: HTTP GET http://localhost:8080/v3/files/v3/files/puppetlabs-apt-1.8.0.tar.gz
Debug: Executing 'gzip -dc /Users/myuser/.puppet/var/puppet-module/cache/puppetlabs-apt20150417-52237-1w8aa5b | tar xof -'
Debug: Executing 'find . -type d -exec chmod 755 {} +'
Debug: Executing 'find . -type f -exec chmod a-wst {} +'
Debug: Executing 'chown -R 1018876844:1884119166 .'
Debug: HTTP GET http://localhost:8080/v3/files/v3/files/puppetlabs-stdlib-4.6.0.tar.gz
Debug: Executing 'gzip -dc /Users/myuser/.puppet/var/puppet-module/cache/puppetlabs-stdlib20150417-52237-mjolpu | tar xof -'
Debug: Executing 'find . -type d -exec chmod 755 {} +'
Debug: Executing 'find . -type f -exec chmod a-wst {} +'
Debug: Executing 'chown -R 1018876844:1884119166 .'
Notice: Installing -- do not interrupt ...
/Users/myuser/.puppet/modules
└─┬ puppetlabs-rabbitmq (v5.1.0)
  ├── nanliu-staging (v1.0.3)
  ├── puppetlabs-apt (v1.8.0)
  └── puppetlabs-stdlib (v4.6.0)
```
